src/App.jsx

```diff
@@ -1,43 +1,26 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import Result from "./Components/Result";
-import './App.css'
+import "./App.css";
 
-class App extends Component {
-  static defaultProps = {
-    randomNum: Math.floor(Math.random() * 100) + 1,
-  };
+const App = () => {
+  const randomNum = Math.floor(Math.random() * 100) + 1;
+  const [userNum, setUserNum] = useState("");
```

Comment:
Can we instead use a number instead of a string for the type of this useState? You can parse the string input value to be an actual number by using the built-in Number class: `Number(event.target.value)`. Also be careful with === vs == in Result.jsx for stuff like this. As we're dealing with a number, we should always try to keep it as a number in the code, even if for a moment it's a string when the user inputs it. To handle non-number input from the user like letters, I suggest you first extract out the onChange callback on the input and define the callback function passed to it above the returned JSX. Then in this callback function first try parsing the input via Number(). If it returns NaN (not a number), then can show your error message. If it's not NaN, then call setUserNum() to set it with the value type being as a number, not a string.